### PR TITLE
feat(24.04): add slices for curl and its dependencies

### PR DESCRIPTION
**Pull request by yashashwi**

### CODE REVIEW

**Comment by yashashwi** [comment]
This PR should now unblock https://github.com/canonical/chisel-releases/pull/107

File: slices/libp11-kit0.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package: libp11-kit0
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+      - libffi8_libs    
+    contents:
+      /lib/*-linux-*/libp11-kit.so.0*:
```
Comment (rebornplusplus) [inline slices/libp11-kit0.yaml:9]:
```suggestion
      /usr/lib/*-linux-*/libp11-kit.so.0*:
```

File: slices/libtasn1-6.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package: libtasn1-6
+
+slices:
+  libs:
+    essential:
+      - libc6_libs   
+    contents:
+      /lib/*-linux-*/libtasn1.so.6*:  {arch: amd64}
```
Comment (rebornplusplus) [inline slices/libtasn1-6.yaml:8]: Why are we extracting it only for `arch: amd64`?

Comment (rebornplusplus) [inline slices/libtasn1-6.yaml:8]:
```suggestion
      /usr/lib/*-linux-*/libtasn1.so.6*:
```

**Review by rebornplusplus — CHANGES_REQUESTED** [review]
Went through it briefly. Please re-check the paths and why are we installing some paths only for a specific arch?

File: slices/libtasn1-6.yaml (hunk 2 above)
Comment (yashashwi) [reply slices/libtasn1-6.yaml:8]: Last time I checked the https://packages.ubuntu.com/noble/libs/libtasn1-6 the packages was showing only for amd64.

I will update all the packages for 24.04 and 23.10

Comment (rebornplusplus) [reply slices/libtasn1-6.yaml:8]: Oh! Packages for 24.04 are changing a lot recently, so maybe that's the case.

File: slices/libp11-kit0.yaml (hunk 1 above)
Comment (yashashwi) [reply slices/libp11-kit0.yaml]: I have fixed this now.

**Comment by yashashwi** [comment]
@rebornplusplus  All issues resolved for this PR

File: slices/libhogweed6.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+slices:
+  libs:
+    essentials:
+      - libc6_libs
+      - libgmp10_libs
+      - libnettle8_libs    
+
+    contents:
+      /usr/lib/*-linux-*/libhogweed.so.6:
```
Comment (cjdcordeiro) [inline slices/libhogweed6.yaml:12]:
```suggestion
      /usr/lib/*-linux-*/libhogweed.so.6*:
```

File: slices/librtmp1.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  libs:
+    essential:
+      - libc6_libs   
+      - libgmp10_libs
+      - libgnutls30_libs
+      - libhogweed6_libs
+      - libnettle8_libs
+      - zlib1g_libs 
+    contents:
+      /usr/lib/*-linux-*/librtmp.so.1*:
```
Comment (cjdcordeiro) [inline slices/librtmp1.yaml:14]:
```suggestion
      /usr/lib/*-linux-*/librtmp.so.1: 
```

File: slices/libsasl2-2.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿package: libsasl2-2
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+      - libsasl2-modules-db_libs
+      - libssl3_libs
+    contents:
+      /usr/lib/*-linux-*/libsasl*.so.2*:
```
Comment (cjdcordeiro) [inline slices/libsasl2-2.yaml:10]:
```suggestion
      /usr/lib/*-linux-*/libsasl2.so.2*: 
```

This is safer, just to make sure it wouldn't conflict with another potential version (e.g. libsasl3)

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
TYVM

Almost there. Just a few adjustments left.

Also, a minor nitpick that would save us some time in the near future would be to lint these yamls. I see some have extra white spaces...some have an empty line between `essentials` and `contents` while others don't. Etc. There's no strict convention (yet), but at least it would be nice if they are linted and consistent :)

File: slices/libsasl2-2.yaml (hunk 5 above)
Comment (yashashwi) [reply slices/libsasl2-2.yaml]: this was a typo, not intentional, fixed it now

**Comment by yashashwi** [comment]
Lint applied, comments addressed.

Fingerscrossed  :)

**Review by cjdcordeiro — APPROVED** [review]
great. LGTM! thanks

**Review by rebornplusplus — APPROVED** [review]
Locally tested the installation of all the slices. Works well, thanks!

**Pull request merged**